opencti-platform/opencti-graphql/src/graphql/sseMiddleware.js

```diff
@@ -190,26 +184,40 @@ const createSeeMiddleware = () => {
       },
       close: () => {
         logApp.info('[STREAM] Closing SSE channel', { clientId: channel.userId });
+        if (heartbeatInterval) clearInterval(heartbeatInterval);
         channel.expirationTime = 0;
-        try {
-          res.end();
-        } catch (e) {
-          logApp.error('[STREAM] Failing to close client', { clientId: channel.userId, error: e });
+        if (!req.finished) {
+          try {
+            res.end();
+          } catch (e) {
+            logApp.error('[STREAM] Failing to close client', { clientId: channel.userId, error: e });
+          }
         }
       },
     };
+    const heartTimer = () => {
```

Comment:
I am not sure to understand this new logic.
Before, we sent Hearthbeat at the end of an action and now, we send it in interval ?

Reply:
Yes thats the point to detached the heartbeat from the processing, to prevent client kill the connection because nothing happens in the stream during long resolution